### DISCUSSION TITLE
Merge different codec files?

### DISCUSSION THREAD

**Discussion by iopq** [mifi/lossless-cut#1203] [Ideas]
Of course this is supposed to be lossless, but I didn't expect my audio to be different, so it actually became garbled. Lossless cut KNOWS this, but still doesn't warn me that the output will be simply garbled. It should ask whether I want to re-encode to another quality, otherwise you can only get garbage either way. I basically only want to re-encode the video that has the different codec, and keep the rest lossless

**Comment by mifi** [comment]
losslesscut doesn't know that the output will be garbled because ffmpeg reports "success" to losslesscut on the operation. Losslesscut doesn't support re-encoding audio for the purpose of concatenating/merging files of different codec, because that's a lossy operation. #547

but I agree it would be nice if it gives a warning, it's just that losslesscut has to contain these heuristics itself, and I have to be careful not to disallow something that is indeed allowed.

**Comment by iopq** [comment]
Is it possible to file an issue over at ffmpeg and see if they can fix just outputting corrupt files? That doesn't seem like the right behavior

**Reply by mifi** [reply]
It's possible but I don't have time to do it. In the next version I'm including a "file check" functionality, where it will show warnings if some codec parameters are different:
<img width="771" alt="Screenshot 2022-06-29 at 16 33 40" src="https://user-images.githubusercontent.com/402547/176563062-d5a0d05a-87f3-4763-92d4-74a238d87eaa.png">

https://github.com/mifi/lossless-cut/blob/fe897b5f4beb9b70f33c1b59c956e76f39ae5378/src/components/ConcatDialog.jsx#L84

**Comment by iopq** [comment]
I guess that's good enough for now, thanks. I've been reencoding with handbrake the files that differ like so:

aac 44.1Khz -> passthrough
aac 48Khz -> reencode to aac 44.1Khz
vorbis -> reencode to aac 44.1Khz

when the video doesn't need reencoding I extract the audio, convert it, and put it back together

then merge the resulting files afterwards (I need to do this for 90 more videos)
